### PR TITLE
Fix: add BindingResult param to api interface template for spring mvc

### PR DESCRIPTION
**Pull request by sophiejanssens**
Added BindingResult method parameter in the interface method for Spring codegen, only used if @valid is used.

### CODE REVIEW

**Review by pa314159 — APPROVED** [review]

File: boat-scaffold/src/main/templates/boat-spring/bodyParams.mustache [hunk 1]
```diff
@@ -2,4 +2,4 @@
...
             {{#useBeanValidation}}
             @Valid{{#required}} @NotNull{{/required}}{{/useBeanValidation}}
-            @RequestBody{{^required}}(required = false){{/required}} {{^reactive}}{{{dataType}}}{{/reactive}}{{#reactive}}{{^isListContainer}}Mono{{/isListContainer}}{{#isListContainer}}Flux{{/isListContainer}}<{{{baseType}}}>{{/reactive}} {{paramName}}{{/isBodyParam}}
\ No newline at end of file
+            @RequestBody{{^required}}(required = false){{/required}} {{^reactive}}{{{dataType}}}{{/reactive}}{{#reactive}}{{^isListContainer}}Mono{{/isListContainer}}{{#isListContainer}}Flux{{/isListContainer}}<{{{baseType}}}>{{/reactive}} {{paramName}} {{#useBeanValidation}}{{#addBindingResult}}, BindingResult result{{/addBindingResult}}{{/useBeanValidation}}{{/isBodyParam}}
```
Comment (pa314159) [inline boat-scaffold/src/main/templates/boat-spring/bodyParams.mustache:5]: There is an extra space before your changes that is not needed

**Review by pa314159 — CHANGES_REQUESTED** [review]
It would be nice to add a test for this new parameter

**Comment by pa314159** [comment]
Please also update README.md
- the Release Notes section
- the list of the Configuration options at the bottom

File: boat-scaffold/src/main/templates/boat-spring/api.mustache [hunk 2]
```diff
@@ -68,6 +69,9 @@ import java.io.IOException;
 {{#addServletRequest}}
 import javax.servlet.http.HttpServletRequest;
 {{/addServletRequest}}
+{{#addBindingResult}}
+import org.springframework.validation.BindingResult
```
Comment (bbzurek) [inline boat-scaffold/src/main/templates/boat-spring/api.mustache:73]: missing `;`

File: README.md [hunk 3]
```diff
@@ -626,6 +626,7 @@ For the `spring` generator, the additional configuration options are:
 | `useClassLevelBeanValidation` | Adds @Validated annotation to API interfaces (Default: false) |
 | `useLombokAnnotations` | Use Lombok annotations to generate properties accessors and `hashCode`/`equals` methods (Default: false) |
 | `addServletRequest` | Adds ServletRequest objects to API method definitions (Default: false) |
+| `addBindingResult` | Adds BindingResult to Api method definitions' request bodies, if UseBeanValidation configured, for this to be effective use configuration UseBeanValidation (Default: false)|
```
Comment (christopherm-backbase) [inline README.md:629]: Does addBindingResult also enable useBeanValidation? The description here is a bit awkward.

**Review by christopherm-backbase — APPROVED** [review]

**Review by pa314159 — APPROVED** [review]

**Pull request merged**